### DISCUSSION TITLE
Is 'aspell' really necessary for PHP?

### DISCUSSION THREAD

**Discussion by techieshark** [Homebrew/discussions#1782] [Everyday usage] [answered]
I've just installed PHP7.4 on a Mac, and noticed that it includes the `aspell` spelling library, which is huge (322MB):
```
==> Installing php dependency: aspell
==> Pouring aspell--0.60.8.big_sur.bottle.1.tar.gz
🍺  /usr/local/Cellar/aspell/0.60.8: 767 files, 322.7MB
```

This is large relative to PHP itself (74MB):
```
/usr/local/Cellar/php@7.4/7.4.21: 497 files, 72.4MB
```

I'm curious if:

a) anyone knows to what extent the `pspell` PHP library is really used (it is what uses aspell under the hood),

b) anyone knows of an easy way to avoid installing aspell while using `brew`'s compiled packages (what I've done now is `brew remove --ignore-dependencies aspell` and will hope PHP still works), and

c) anyone knows how off hand how PHP can be compiled from source using brew such that aspell is not installed, and

d) perhaps `brew` could come with aspell enabled by default, but have it able to be installed and enabled as a separate module. 

That's a lot of questions for one package! If anyone has info on even just one of those things I'd be appreciative. Thanks!

**Comment by SMillerDev** [comment, marked as answer]
> a) anyone knows to what extent the pspell PHP library is really used (it is what uses aspell under the hood),

It's part of the PHP source so potentially as much as PHP itself.

> b) anyone knows of an easy way to avoid installing aspell while using brew's compiled packages (what I've done now is brew remove --ignore-dependencies aspell and will hope PHP still works), and

Maintain your own version of PHP that does not include aspell.

> c) anyone knows how off hand how PHP can be compiled from source using brew such that aspell is not installed, and

`brew extract` is the command you're looking for. 

> d) perhaps brew could come with aspell enabled by default, but have it able to be installed and enabled as a separate module.

Since people expect from brew installs that it "just works" this would be very unexpected behavior for brew.

**Reply by techieshark** [reply]
Thanks! I might give that `brew extract` a try and an create a custom slim php package.
